pennylane/math/quantum.py

```diff
@@ -1032,3 +1032,83 @@ def _check_state_vector(state_vector):
     if not is_abstract(norm):
         if not allclose(norm, 1.0, atol=1e-10):
             raise ValueError("Sum of amplitudes-squared does not equal one.")
+
+
+def max_entropy(state, indices, base=None, check_state=False, c_dtype="complex128"):
+    r"""Compute the maximum entropy from a state vector or density matrix on a given subsystem. It supports all
+    interfaces (Numpy, Autograd, Torch, Tensorflow and Jax).
+
+    .. math::
+        S( \rho )_0 = \log( \text{rank} ( \rho ))
+
+    Args:
+        state (tensor_like): ``(2**N)`` state vector or ``(2**N, 2**N)`` density matrix.
+        indices (list(int)): List of indices in the considered subsystem.
+        base (float): Base for the logarithm. If None, the natural logarithm is used.
+        check_state (bool): If True, the function will check the state validity (shape and norm).
+        c_dtype (str): Complex floating point precision type.
+
+    Returns:
+        float: The maximum entropy of the considered subsystem.
+
+    **Example**
+
+    The maximum entropy of a subsystem for any state vectors can be obtained. Here is an example for the
+    maximally entangled state, where the subsystem entropy is maximal (default base for log is exponential).
+
+
+    >>> x = [1, 0, 0, 1] / np.sqrt(2)
+    >>> max_entropy(x, indices=[0])
+    0.6931472
+
+    The logarithm base can be switched to 2 for example.
+
+    >>> max_entropy(x, indices=[0], base=2)
+    1.0
+
+    The maximum entropy can be obtained by providing a quantum state as a density matrix, for example:
+
+    >>> y = [[1/2, 0, 0, 1/2], [0, 0, 0, 0], [0, 0, 0, 0], [1/2, 0, 0, 1/2]]
+    >>> max_entropy(x, indices=[0])
+    0.6931472

```

Comment:
Nice examples! Could you add a small comparison with a VN entropy value and show that the maximal entropy is always greater?

Reply:
Thanks! I added ann additional example comparing with VN entropy. Let me know what you think